pkg/api/query/v1.go

```diff
@@ -120,7 +144,21 @@ func (f *QueryEngineFactory) GetThanosEngine() promql.QueryEngine {
 		if f.remoteEngineEndpoints == nil {
 			f.thanosEngine = engine.New(engine.Opts{EngineOpts: f.engineOpts, Engine: f.GetPrometheusEngine(), EnableAnalysis: true, EnableXFunctions: f.enableXFunctions})
 		} else {
-			f.thanosEngine = engine.NewDistributedEngine(engine.Opts{EngineOpts: f.engineOpts, Engine: f.GetPrometheusEngine(), EnableAnalysis: true}, f.remoteEngineEndpoints)
+			f.thanosEngine = newSecondPrecisionEngine(
```

Comment:
Hm I don't think we need this change. The distributed engine already truncates milliseconds.